Src/HALALMock/Services/Communication/Ethernet/Ethernet.cpp

```diff
@@ -88,57 +39,41 @@ void Ethernet::start(IPV4 local_ip, IPV4 subnet_mask, IPV4 gateway){
 		GATEWAY_ADDRESS[1] = (gw.addr >> 8) & 0xFF;
 		GATEWAY_ADDRESS[2] = (gw.addr >> 16) & 0xFF;
 		GATEWAY_ADDRESS[3] = (gw.addr >> 24) & 0xFF;
-		MX_LWIP_Init();
 		is_running = true;
 	}else{
-		ErrorHandler("Unable to start Ethernet!");
+		std::cout<<"Unable to start Ethernet!"<<std::endl;
 	}
 
 	if (not is_ready) {
-		ErrorHandler("Ethernet is not ready");
+		std::cout<<"Ethernet is not ready"<<std::endl;
 		return;
 	}
 
 }
 
 void Ethernet::inscribe(){
 	if(!is_ready){
-		Pin::inscribe(PA1, ALTERNATIVE);
-		Pin::inscribe(PA2, ALTERNATIVE);
-		Pin::inscribe(PA7, ALTERNATIVE);
-		Pin::inscribe(PB13, ALTERNATIVE);
-		Pin::inscribe(PC1, ALTERNATIVE);
-		Pin::inscribe(PC4, ALTERNATIVE);
-		Pin::inscribe(PC5, ALTERNATIVE);
-		Pin::inscribe(PG11, ALTERNATIVE);
-		Pin::inscribe(PG0, ALTERNATIVE);
-		Pin::inscribe(PG13, ALTERNATIVE);
-		is_ready = true;
+		uint8_t number_pin_ethernet = 10;
+		Pin pin_list_ethernet[number_pin_ethernet] = {PA1,PA2,PA7,PB13,PC1,PC4,PC5,PG11,PG0,PG13};
+		for(size_t i = 0; i < number_pin_ethernet; i++){
+			EmulatedPin &pin_data = SharedMemory::get_pin(pin_list_ethernet[i]);
+			if(pin_data.type == PinType::NOT_USED){
+			pin_data.type = PinType::Ethernet;
+			}else{
+				std::cout<<"Error inscribing ethernet Pins, PA1,PA2,PA7,PB13,PC1,PC4,PC5,PG11,PG0,PG13 must be free"<<std::endl;
+				return;
+			}
+		}
```

Comment:
I'm not sure if we have to do this, but it doesn't hurt

Reply:
I think if some pins are being used in ethernet we should, have them in the gpio_memory so if someone try to access for a pwm they receive an error message